File: src/converter/plugins/convertQualifiedName.ts
import ts from "typescript";
import {createSimplePlugin} from "../plugin.js";
import {CheckCoverageService, checkCoverageServiceKey} from "./CheckCoveragePlugin.js";

export const convertQualifiedName = createSimplePlugin((node, context, render) => {
    if (!ts.isQualifiedName(node)) return null

    const checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    return `${render(node.left)}.${render(node.right)}`
})

File: karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/plugins/convertQualifiedName.kt
package io.github.sgrishchenko.karakum.extension.plugins

import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isQualifiedName

val convertQualifiedName = createPlugin plugin@{ node, context, render ->
    if (!isQualifiedName(node)) return@plugin null

    val checkCoverageService = context.lookupService(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    "${render(node.left)}.${render(node.right)}"
}
